algebird-core/src/main/scala/com/twitter/algebird/HyperLogLog.scala

```diff
@@ -43,6 +43,8 @@ object HyperLogLog {
   /* Size of the hash in bits */
   val hashSize = 128
 
+  val powersOfNegativeTwo: Array[Double] = 0.until(hashSize).map{ i => math.pow(2.0, -i) }.toArray
```

Comment:
Cool.  For the record the parts that look not right are in rhoW:
- terminal condition is that `contains` returns false, but contains doesn't do any bounds checking; is it possible to overrun the array, however unlikely?
- loop starts at 1 instead of 0.  with even a small aggregation I imagine this doesn't matter, but isn't it technically incorrect?

Again I could be missing something, but if you think they might be worth addressing maybe lets file issues.